include/core/boundary_conditions.h

```diff
@@ -449,24 +517,36 @@ namespace BoundaryConditions
   class HTBoundaryConditions : public BoundaryConditions<dim>
   {
   public:
-    std::vector<std::shared_ptr<Functions::ParsedFunction<dim>>>
+    std::map<types::boundary_id,
+             std::shared_ptr<Functions::ParsedFunction<dim>>>
       dirichlet_value;
-    std::vector<std::shared_ptr<Functions::ParsedFunction<dim>>> h;
-    std::vector<std::shared_ptr<Functions::ParsedFunction<dim>>> Tinf;
-    std::vector<std::shared_ptr<Functions::ParsedFunction<dim>>> emissivity;
-    std::vector<std::shared_ptr<Functions::ParsedFunction<dim>>> heat_flux_bc;
+    std::map<types::boundary_id,
+             std::shared_ptr<Functions::ParsedFunction<dim>>>
+      h;
+    std::map<types::boundary_id,
+             std::shared_ptr<Functions::ParsedFunction<dim>>>
+      Tinf;
+    std::map<types::boundary_id,
+             std::shared_ptr<Functions::ParsedFunction<dim>>>
+      emissivity;
+    std::map<types::boundary_id,
+             std::shared_ptr<Functions::ParsedFunction<dim>>>
+           heat_flux_bc;
     double Stefan_Boltzmann_constant;
 
     void
-    declareDefaultEntry(ParameterHandler &prm, const unsigned int i_bc);
+    declare_default_entry(ParameterHandler        &prm,
+                          const types::boundary_id default_boundary_id);
     void
     declare_parameters(ParameterHandler  &prm,
                        const unsigned int number_of_boundary_conditions);
     void
-    parse_boundary(ParameterHandler &prm, const unsigned int i_bc);
+    parse_boundary(ParameterHandler &prm);
     void
     parse_parameters(ParameterHandler &prm);
 
+
+    // TODO REMOVE THIS FLAG IT IS NOT USEFUL
```

Comment:
Did you want to remove this flag now or in another PR?

Reply:
Maybe in another PR. This PR is already kinda bloated in my opinion..